cpp/include/cudf_test/column_wrapper.hpp

```diff
@@ -202,6 +208,46 @@ rmm::device_buffer make_elements(InputIterator begin, InputIterator end)
   return rmm::device_buffer{elements.data(), size * sizeof(RepType), rmm::cuda_stream_default};
 }
 
+#ifdef __CUDACC__
```

Comment:
An alternative approach here is to move all the `__CUDACC__` surrounded code into its own header file (e.g. `column_wrapper.cuh`) and have that file (.cuh) include this one (.hpp). This way, the `.cuh` is included in `.cu` files and the `.hpp` file is included in `.cpp` files. A `.cpp` file will not call any of the functions in the new `.cuh` file anyway.
I was able to get this working locally so I can share this solution if you want.

Reply:
That is a ODR violation. The class `column_wrapper` will have a different definition in a nvcc TU vs a non-nvcc TU.

Reply:
The declaration will be the same and the definition will only differ where the `#ifdef __CUDACC__` makes it different anyway.

Reply:
@jrhemstad Do you have further thoughts on this? I would like to better understand if/how the `#ifdef __CUDACC__` approach is different with respect to ODR violations than the `.cuh` approach proposed by @davidwendt.

Reply:
Removed these new member functions and added code withing old member function to make decision on call device code all based on iterator type and `#ifdef __CUDACC__`. 
device code cannot be compiled with cpp only files, the code is dependent on template Iterator type, so `#ifdef __CUDACC__` is not avoidable.

Does this present code still violate One Definition Rule?